src/platform_util.cpp

```diff
@@ -73,9 +73,9 @@ namespace libtorrent
 		if (getrlimit(RLIMIT_NOFILE, &rl) == 0)
 		{
 			if (rl.rlim_cur == rlim_infinity)
-				return (std::numeric_limits<int>::max)();
+				return std::numeric_limits<int>::max();
 
-			return rl.rlim_cur;
+			return int(rl.rlim_cur);
```

Comment:
this feels like something that could easily overflow. would you mind making sure that if ``rlim_cur`` > ``INT_MAX``, that ``INT_MAX`` is returned?